### DISCUSSION TITLE
How to generate SQLite files using data loaders?

### DISCUSSION THREAD

**Discussion by Thomascountz** [observablehq/framework#1266] [Q&A] [answered]
My SQLite workflow is to execute an external script and copy the `.sqlite` file into `docs/data` in order to use `observablehq/sqlite` via `FileAttachment`.

However, I'm curious about how I could take advantage of the automation provided by data loaders:

> Using FileAttachment means that referenced files are automatically copied to dist during build, and you can even generate SQLite files using data loaders. 
> https://observablehq.com/framework/lib/sqlite

It has worked to use an I/O stream to write the file binary to STDOUT, for example:

```ruby
# docs/data/people.sqlite.rb

# Create or open the database
DB_FILENAME = 'people.sqlite'
db = SQLite3::Database.new(DB_FILENAME)

# Create a table if it doesn't exist
db.execute <<-SQL
  CREATE TABLE IF NOT EXISTS people (
    id INTEGER PRIMARY KEY,
    name TEXT,
    age INTEGER
  );
SQL

# Insert a few rows of data
db.execute "INSERT INTO people (name, age) VALUES (?, ?)", ["Alice", 30]
db.execute "INSERT INTO people (name, age) VALUES (?, ?)", ["Bob", 45]
db.execute "INSERT INTO people (name, age) VALUES (?, ?)", ["Charlie", 25]
db.close

# Write the database file to STDOUT via an IO stream
IO.copy_stream(open(DB_FILENAME, 'rb'), $stdout)
```

I'm only wondering if the documentation is alluding to a different way of generating SQLite files via data loaders?

**Comment by Fil** [comment, marked as answer]
That seems like a good way of generating sqlite files!
